### PR TITLE
[BEAM-8133] Update influx db publisher

### PR DESCRIPTION
**Pull request by pawelpasterz**
`runtimeMs` and `numResults` are now published as integer not float type
Additional logic for error logging added

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark
--- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.
H

### CODE REVIEW

**Comment by pawelpasterz** [comment]
R: @kamilwu 
Please, take a look. Thanks!

File: sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Main.java [hunk 1]
```diff
@@ -255,7 +255,9 @@ private static String produceMeasurement(
     schemaResults.put("timestamp", timestamp);
     schemaResults.put("runner", runner);
     schemaResults.put("measurement", measurement);
-    schemaResults.put("runtimeMs", runtimeMs);
+    schemaResults.put(
+        "runtimeMs",
+        runtimeMs + "i"); // we need to add 'i' suffix since influx default type is float
```
Comment (kamilwu) [inline sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Main.java:260]: What about something like this?
`
By default, InfluxDB treats all values as floats. We need to add 'i' suffix to interpret the value as an integer.
`

Comment (pawelpasterz) [reply sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Main.java]: Fine for me, updated

File: sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/InfluxDBPublisher.java [hunk 2]
```diff
@@ -146,16 +157,29 @@ private static HttpPost providePOSTRequest(final InfluxDBSettings settings) {
...
+    final int code = response.getStatusLine().getStatusCode();
     if (code < 200 || code >= 300) {
-      throw new IOException("Response code: " + code);
+      throw new IOException(
+          "Response code: " + code + ". Reason: " + getErrorMessage(response.getEntity()));
     }
   }
 
+  private static String getErrorMessage(final HttpEntity entity) throws IOException {
+    final Header encodingHeader = entity.getContentEncoding();
```
Comment (kamilwu) [inline sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/InfluxDBPublisher.java:173]: If you want to make it even more secure and resilient, you can check what's the value of `Content-Type` header. For JSON, it must be `application/json`. Some HTTP libraries has convenient methods for doing that.

Comment (pawelpasterz) [reply sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/InfluxDBPublisher.java:173]: Yes, that's why I check encoding type to be prepared for case when influx would stop sending us JSON (and start something else)
```
final Charset encoding =
        encodingHeader == null
            ? StandardCharsets.UTF_8
            : Charsets.toCharset(encodingHeader.getValue());
```

**Comment by kamilwu** [comment]
Run Direct Runner Nexmark Tests

**Comment by kamilwu** [comment]
Retest this please

**Comment by kamilwu** [comment]
Run Java PreCommit

File: sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/InfluxDBPublisher.java (hunk 2 above)
Comment (kamilwu) [reply sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/InfluxDBPublisher.java:173]: Isn't it possible that you get UTF-8-encoded data, but not necessarily JSON?

File: sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Main.java [hunk 3]
```diff
@@ -255,7 +255,11 @@ private static String produceMeasurement(
     schemaResults.put("timestamp", timestamp);
     schemaResults.put("runner", runner);
     schemaResults.put("measurement", measurement);
-    schemaResults.put("runtimeMs", runtimeMs);
+    schemaResults.put(
+        "runtimeMs",
```
Comment (kamilwu) [inline sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Main.java:259]: We should do the same for `numResults`. Now the schema that is being created is:
```
fieldKey   fieldType
--------   ---------
numResults float
runtimeMs  integer
```
And we need `numResults` as an integer too. I'm sorry I didn't notice that earlier.

**Comment by kamilwu** [comment]
Run Direct Runner Nexmark Tests

**Comment by kamilwu** [comment]
Run Seed Job

**Comment by kamilwu** [comment]
Run Direct Runner Nexmark Tests

**Comment by kamilwu** [comment]
LGTM, thanks

**Pull request merged**